gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -241,14 +241,18 @@ public Boolean load(String bucketName) throws Exception {
    */
   public GoogleCloudStorageImpl(GoogleCloudStorageOptions options, Credential credential)
       throws IOException {
-    this(options, new RetryHttpInitializer(credential, options.toRetryHttpInitializerOptions()));
+    this(
+        options,
+        createStorage(
+            options, new RetryHttpInitializer(credential, options.toRetryHttpInitializerOptions())),
+        credential);
```

Comment:
Instead of adding this constructor and duplicating `credential` that already passed in the `RetryHttpInitializer` may you add a package-scoped getter in the `RetryHttpInitializer` class for `credential` field?

Reply:
Done.